hazelcast/src/test/java/com/hazelcast/map/impl/nearcache/NearCacheTest.java

```diff
@@ -229,6 +230,188 @@ public void testNearCacheEviction_withMapClear() {
+
+        // populate map
+        IMap<Integer, Integer> map = instances[0].getMap(mapName);
+        populateMap(map, mapSize);
+
+        // populate Near Cache
+        populateNearCache(map, mapSize);
+
+        NearCacheStats stats = getNearCacheStats(map);
+
+        // make some hits
+        populateNearCache(map, mapSize);
+
+        long invalidationsBefore = stats.getInvalidations();
+        Set<Integer> set = new HashSet<Integer>();
+        for (int i = 0; i < mapSize; i++) {
+            set.add(i);
+        }
+        map.submitToKeys(set, new TestReadOnlyProcessor());
+        long invalidationsAfter = stats.getInvalidations();
+        assertEquals("There should be no invalidation after getting keys from read only entry processor", invalidationsBefore, invalidationsAfter);
+    }
+
+    @Test
+    public void testNearCacheStatsWithReadOnlyProcessor_executeOnEntriesWithPredicate() {
+        int mapSize = 1000;
+        String mapName = randomMapName();
+
+        Config config = getConfig();
+        config.getMapConfig(mapName).setNearCacheConfig(newNearCacheConfig().setInvalidateOnChange(false));
+
+        TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(2);
+        HazelcastInstance[] instances = factory.newInstances(config);
+
+        IMap<Integer, Employee> map = instances[0].getMap(mapName);
+        for (int i = 0; i < mapSize; i++) {
+            map.put(i, new Employee(i, "", 0, true, 0D));
+        }
+
+        populateNearCache(map, mapSize);
+
+        NearCacheStats stats = getNearCacheStats(map);
+
+        long invalidationsBefore = stats.getInvalidations();
+        EntryObject e = Predicates.newPredicateBuilder().getEntryObject();
+        Predicate predicate = e.get("salary").equal(0);
+        map.executeOnEntries(new TestObjectBasedReadOnlyProcessor(), predicate);
+        long invalidationsAfter = stats.getInvalidations();
+        assertEquals("There should be no invalidation after getting keys from read only entry processor", invalidationsBefore, invalidationsAfter);
+    }
+
+    @Test
+    public void testNearCacheStatsWithReadOnlyProcessor_executeOnEntriesWithoutPredicate() {
+        int mapSize = 1000;
+        String mapName = randomMapName();
+
+        Config config = getConfig();
+        config.getMapConfig(mapName).setNearCacheConfig(newNearCacheConfig().setInvalidateOnChange(false));
+
+        TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(2);
+        HazelcastInstance[] instances = factory.newInstances(config);
+
+        IMap<Integer, Employee> map = instances[0].getMap(mapName);
+        for (int i = 0; i < mapSize; i++) {
+            map.put(i, new Employee(i, "", 0, true, 0D));
+        }
+
+        populateNearCache(map, mapSize);
+
+        NearCacheStats stats = getNearCacheStats(map);
+
+        long invalidationsBefore = stats.getInvalidations();
+        EntryObject e = Predicates.newPredicateBuilder().getEntryObject();
```

Comment:
Unused variable. I think you should use integer based map, remove TestObjectBasedReadOnlyProcessor and reuse the method you will introduce for the logic that will setup instances, populate near cache etc.